agogosml/agogosml/common/kafka_streaming_client.py

```diff
@@ -53,12 +50,24 @@ def create_kafka_config(user_config: dict) -> dict:
         config = {
             "bootstrap.servers": user_config.get("KAFKA_ADDRESS"),
             "enable.auto.commit": False,
-            "auto.offset.reset": "earliest"
+            "auto.offset.reset": "earliest",
+            "default.topic.config": {'auto.offset.reset': 'smallest'},
         }
 
-        consumer_group = user_config.get("KAFKA_CONSUMER_GROUP")
-        if consumer_group is not None:
-            config["group.id"] = consumer_group
+        if user_config.get('EVENTHUB_KAFKA_CONNECTION_STRING'):
+            ssl_location = user_config.get('SSL_CERT_LOCATION') or '/etc/ssl/ca-certificates.crt'
+            eventhub_config = {
+                'security.protocol': "SASL_SSL",
+                'sasl.mechanism': "PLAIN",
+                'ssl.ca.location': ssl_location,
+                'sasl.username': '$ConnectionString',
+                'sasl.password': user_config.get('EVENTHUB_KAFKA_CONNECTION_STRING'),
+                'client.id': 'agogosml',
+            }
+            config = config.update(eventhub_config)
+
+        if user_config.get("KAFKA_CONSUMER_GROUP") is not None:
```

Comment:
Replace with:

```py
if 'KAFKA_CONSUMER_GROUP' in user_config:
  config['group.id'] = user_config['KAFKA_CONSUMER_GROUP']
```

The result of `dict.get` is an optional and the result `dict[...]` is always a value (or an exception is raised) which should make the type checker shut up here.